dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -161,51 +120,22 @@ export default defineComponent({
     /**
      * Handles selection of a file by the user. First it checks if the file name is already taken.
      * If yes, the selected file is removed again and a popup with an error message is shown.
-     * Else the file is added to the reports that shall be uploaded, then the sha256 hashes are calculated and added
-     * to the respective files.
-     * @param event full event object containing the files
-     * @param event.files files
+     * Else the file is added to the reports that shall be uploaded, then the sha256 hashes are calculated
+     * and added to the respective files.
+     * @param reports the list of all reports currently selected in the file upload
      */
-    async handleFilesSelected(event: FileUploadSelectEvent): void {
-      const selectedFilesByUser = event.files as File[];
-      if (!this.isThereActuallyANewFileSelected(selectedFilesByUser, this.reportsToUpload)) {
-        return;
-      }
-      const indexOfLastSelectedFile = selectedFilesByUser.length - 1;
-      const lastSelectedFile = selectedFilesByUser[indexOfLastSelectedFile];
-      if (this.isFileNameAlreadyExistingAmongReferenceableReportNames(lastSelectedFile.name)) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-        this.$refs.fileUpload.remove(indexOfLastSelectedFile);
+    updateSelectedReports(reports: ReportToUpload[]) {
+      this.reportsToUpload = reports;
+      if (this.duplicatesAmongReferenceableReportNames()) {
+        const indexOfLastSelectedFile = reports.length - 1;
+        const lastSelectedFile = reports[indexOfLastSelectedFile].file;
         this.openModalToDisplayDuplicateNameError(lastSelectedFile.name);
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```

Comment:
Can this eslint disable be removed?

Reply:
It can, as long as sonar does not complain. The typescript linter is very unhappy with this but we are using this kind of calling child components on several occasions and think that it is very useful.

Reply:
did fix it :) CI is green now except for the keycloak false negative code smell
as stated in the ticket and proposed in the daily, i will try to fix the other occasions of $ref + eslint-disable in a similar way